template/golang-http/go.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+# go.sh is a wrapper for the Go command that will
+# automatically set the required module related flags
+# when a vendor folder is detected.
+#
+# Currently, in Go 1.18, Go Workspaces are incompatible
```

Comment:
will be a dumb question but if this was 1.17 would we need this tweak?

I am kinda thinking how all sh scripts can be avoided but I thought vendor files were compatible with workspaces, it is intriguing. I think http templates are super simple compared to past sh scripts before 1.18 and this amount seems OK, I will chase this incompatibility issue

Reply:
actually -mod readonly also reads the vendor as far as I understand from this thread https://github.com/golang/go/issues/30404, so I think shell script may no longer needed? the end user needs to do -mod=readonly as you pointed out here Lucas https://github.com/LucasRoesler/openfaas-golang-template-workspace

Reply:
@MrWormHole i don't 100% follow. Were you able to get a workspaces project to build with the vendor by using `-mod=vendor -mod=readonly`?

The main reason for the script, is that when using workspaces, `go build` ignores the vendor folder and attempts to download the modules again. When a vendor folder is detected, disabling modules forces Go to use the GOPATH, which then, correctly, detects and uses the vendor folder as expected.

Reply:
-mod=readonly should use vendor if vendor exists, I also don't know why this config needs to specify -mod=vendor to get vendor files, it is automatic when not specified, default is -mod=readonly, -mod=mod is the one who only looks up for go modules and ignores vendor completely

Reply:
I understand what you are saying, but 

1. this code does not use or require `-mod=vendor`
2. you are 100% right, we normally expect `go build` to automatically detect the `vendor` folder, however, the current behavior of _workspaces_ seems to ignore this.  I tried it (I had exactly the same hope that you did) but it was not building in the correct way. Even with a `vendor` folder present it was still downloading the modules. 


Perhaps you can share a working example repo with the changes you would like to see? 

When I tried to use vendor with the `go.work`, the only time I could get it to detect the `vendor` folder was by disabling modules. This is what the script does. It automates detecting this situation instead of requiring the developer to pass an additional build arg.

Reply:
so I actually created the same repository but without -mod vendor flag, I vendored my dependencies (which is the user responsibility) and deleted -mod=vendor and made go modules ON, it actually detected and built with vendor folder

Reply:
If we can move forward without using additional bash and if statements, I'd be interested to see that.

Reply:
I think Lucas is right @LucasRoesler depending on your Go version it gets set to default to -mod=readonly, I would keep it and resolve these comments. It will enhance the dev usage of these templates even though it sets it internally.

https://go.dev/ref/mod
![image](https://user-images.githubusercontent.com/22800416/176490527-31676813-aa91-4a21-ba09-c80249803df6.png)

Reply:
@MrWormHole can you clarify your recommendations in a few bullets or steps, or are you retracting what you said?